### PR TITLE
feat / ui improvements

### PR DESCRIPTION
**Pull request by yzhang1994**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue: https://github.com/CoinAlpha/hummingbot/issues/35 and https://github.com/CoinAlpha/hummingbot/issues/135
- Related Clubhouse Story:
https://app.clubhouse.io/coinalpha/story/3501/add-symbol-validator
https://app.clubhouse.io/coinalpha/story/3702/change-keyboard-shortcuts-for-exit-and-copy
https://app.clubhouse.io/coinalpha/story/3785/add-autocomplete-for-import-and-create
https://app.clubhouse.io/coinalpha/story/3788/improve-cli-ease-of-use

**A description of the changes proposed in the pull request**:
- Add symbol validator (Invalid exchange symbols will not be accepted)
- Change exit shortcut to Double CTRL + C
- Add autocomplete for input options (such as `import/create`)
- Made path completer only search in `conf/` directory
- Autofill input field with "start" after config step is complete
- Add CTRL + S hotkey for `status()`

### CODE REVIEW

File: hummingbot/cli/ui/completer.py [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-    List,
-    Dict,
-)
+import re
+from typing import List
 from prompt_toolkit.completion import Completer, WordCompleter, PathCompleter, CompleteEvent
 from prompt_toolkit.document import Document
 
-from hummingbot.cli.settings import EXCHANGES, STRATEGIES
+from hummingbot.cli.settings import EXCHANGES, STRATEGIES, CONF_FILE_PATH, symbol_fetcher
```
Comment (martinkou) [inline hummingbot/cli/ui/completer.py]: Please use the new multi-line import style.

Comment (yzhang1994) [reply hummingbot/cli/ui/completer.py]: fixed

File: hummingbot/cli/hummingbot_application.py [hunk 2]
```diff
@@ -226,7 +226,7 @@ def config(self, key: str = None):
 
     async def _unlock_wallet(self):
         choice = await self.app.prompt(prompt="Would you like to unlock your previously saved wallet? (y/n) >>> ")
-        if choice.lower() == "y":
+        if choice.lower() in {"", "y", "yes"}:
```
Comment (chowryan) [inline hummingbot/cli/hummingbot_application.py]: easy for someone to accidentally mash on enter repeatedly and accidentally confirm?

File: hummingbot/cli/hummingbot_application.py [hunk 3]
```diff
@@ -684,7 +685,7 @@ def start(self, log_level: Optional[str] = None):
 
             ans = await self.app.prompt("Are you sure you want to print your private key in plain text? (y/n) >>> ")
 
-            if ans.lower() in {"y" or "yes"}:
+            if ans.lower() in {"", "y", "yes"}:
```
Comment (chowryan) [inline hummingbot/cli/hummingbot_application.py]: same as above

File: hummingbot/cli/settings.py [hunk 4]
```diff
@@ -153,9 +157,36 @@ def is_strategy(value: str) -> bool:
...
 
 def is_path(value: str) -> bool:
-    return isfile(value) and value.endswith('.yml')
+    return isfile(join(CONF_FILE_PATH, value)) and value.endswith('.yml')
+
+
+def is_valid_market_symbol(market: str, value: str) -> bool:
+    if symbol_fetcher.ready:
+        market_symbols = symbol_fetcher.symbols.get(market, [])
+        return value in symbol_fetcher.symbols.get(market) if len(market_symbols) > 0 else True
```
Comment (chowryan) [inline hummingbot/cli/settings.py]: this should default to False?

Comment (chowryan) [inline hummingbot/cli/settings.py]: no return if symbol_fetcher is not ready

Comment (chowryan) [reply hummingbot/cli/settings.py]: maybe add a comment explaining why default is true

File: hummingbot/cli/ui/completer.py [hunk 5]
```diff
@@ -99,6 +111,7 @@ def get_completions(self, document: Document, complete_event: CompleteEvent):
         if self._complete_paths(document):
             for c in self._path_completer.get_completions(document, complete_event):
                 yield c
+            return
```
Comment (chowryan) [inline hummingbot/cli/ui/completer.py]: is this return intentional?

Comment (yzhang1994) [reply hummingbot/cli/ui/completer.py]: yes

File: hummingbot/cli/utils/symbol_fetcher.py [hunk 6]
```diff
@@ -8,73 +8,90 @@
...
-                    symbol_structs = data.get("symbols")
-                    symbols = list(map(lambda symbol_details: symbol_details.get('symbol'), symbol_structs))
-                    return symbols
-                except Exception:
-                    # Do nothing if the request fails -- there will be no autocomplete for binance symbols
-                    return []
+class SymbolFetcher:
+    def __init__(self):
+        self.ready = False
+        self.symbols = {}
```
Comment (chowryan) [inline hummingbot/cli/utils/symbol_fetcher.py]: add typing for self.symbols

File: hummingbot/cli/utils/symbol_fetcher.py [hunk 7]
```diff
@@ -8,73 +8,90 @@
...
+                        return symbols
+                    except Exception:
+                        # Do nothing if the request fails -- there will be no autocomplete for coinbase symbols
+                        return []
 
+    async def fetch_all(self) -> Dict[str, List[str]]:
+        binance_symbols = await self.fetch_binance_symbols()
+        ddex_symbols = await self.fetch_ddex_symbols()
+        radar_relay_symbols = await self.fetch_radar_relay_symbols()
+        coinbase_pro_symbols = await self.fetch_coinbase_pro_symbols()
```
Comment (chowryan) [inline hummingbot/cli/utils/symbol_fetcher.py]: use asyncio.gather?

File: hummingbot/cli/utils/symbol_fetcher.py [hunk 8]
```diff
@@ -8,73 +8,90 @@
...
+            async with client.get(BINANCE_ENDPOINT, timeout=API_CALL_TIMEOUT) as response:
+                if response.status == 200:
+                    try:
+                        data = await response.json()
+                        symbol_structs = data.get("symbols")
+                        symbols = list(map(lambda symbol_details: symbol_details.get('symbol'), symbol_structs))
+                        return symbols
+                    except Exception:
+                        # Do nothing if the request fails -- there will be no autocomplete for binance symbols
+                        return []
```
Comment (chowryan) [inline hummingbot/cli/utils/symbol_fetcher.py]: no return if status is not 200

File: hummingbot/cli/utils/symbol_fetcher.py [hunk 9]
```diff
@@ -8,73 +8,90 @@
...
+            async with client.get(DDEX_ENDPOINT, timeout=API_CALL_TIMEOUT) as response:
+                if response.status == 200:
+                    try:
+                        response = await response.json()
+                        markets = response.get("data").get("markets")
+                        symbols = list(map(lambda symbol_details: symbol_details.get('id'), markets))
+                        return symbols
+                    except Exception:
+                        # Do nothing if the request fails -- there will be no autocomplete for ddex symbols
+                        return []
```
Comment (chowryan) [inline hummingbot/cli/utils/symbol_fetcher.py]: no return if status is not 200

File: hummingbot/cli/utils/symbol_fetcher.py [hunk 10]
```diff
@@ -8,73 +8,90 @@
...
+        async with aiohttp.ClientSession() as client:
+            async with client.get(COINBASE_PRO_ENDPOINT, timeout=API_CALL_TIMEOUT) as response:
+                if response.status == 200:
+                    try:
+                        markets = await response.json()
+                        symbols = list(map(lambda symbol_details: symbol_details.get('id'), markets))
+                        return symbols
+                    except Exception:
+                        # Do nothing if the request fails -- there will be no autocomplete for coinbase symbols
+                        return []
```
Comment (chowryan) [inline hummingbot/cli/utils/symbol_fetcher.py]: no return if status is not 200

File: hummingbot/cli/hummingbot_application.py (hunk 2 above)
Comment (yzhang1994) [reply hummingbot/cli/hummingbot_application.py]: fixed

File: hummingbot/cli/hummingbot_application.py (hunk 3 above)
Comment (yzhang1994) [reply hummingbot/cli/hummingbot_application.py]: fixed

File: hummingbot/cli/utils/symbol_fetcher.py (hunk 6 above)
Comment (yzhang1994) [reply hummingbot/cli/utils/symbol_fetcher.py]: fixed

File: hummingbot/cli/utils/symbol_fetcher.py (hunk 8 above)
Comment (yzhang1994) [reply hummingbot/cli/utils/symbol_fetcher.py]: fixed

File: hummingbot/cli/utils/symbol_fetcher.py (hunk 10 above)
Comment (yzhang1994) [reply hummingbot/cli/utils/symbol_fetcher.py]: fixed

File: hummingbot/cli/utils/symbol_fetcher.py (hunk 9 above)
Comment (yzhang1994) [reply hummingbot/cli/utils/symbol_fetcher.py]: fixed

File: hummingbot/cli/settings.py (hunk 4 above)
Comment (yzhang1994) [reply hummingbot/cli/settings.py]: added

File: hummingbot/cli/utils/symbol_fetcher.py (hunk 7 above)
Comment (yzhang1994) [reply hummingbot/cli/utils/symbol_fetcher.py]: I will refactor this function later to fetch symbols for every exchange.

**Pull request merged**